targets/TARGET_Cypress/TARGET_PSOC6_FUTURE/TARGET_FUTURE_SEQUANA_M0_PSA/spm_init_api.c

```diff
@@ -65,9 +66,12 @@ void mailbox_init(void)
 
     // Configure interrupts ISR / MUX and priority
     cy_stc_sysint_t ipc_intr_Config;
-    ipc_intr_Config.intrSrc = (IRQn_Type)NvicMux3_IRQn; // Can be any Mux we choose
+    ipc_intr_Config.intrSrc = CY_M0_CORE_IRQ_CHANNEL_PSA_MAILBOX;
     ipc_intr_Config.cm0pSrc = (cy_en_intr_t)cpuss_interrupts_ipc_0_IRQn + SPM_IPC_NOTIFY_CM0P_INTR;    // Must match the interrupt we trigger using NOTIFY on CM4
     ipc_intr_Config.intrPriority = 1;
+    if (cy_m0_nvic_reserve_channel(CY_M0_CORE_IRQ_CHANNEL_PSA_MAILBOX, CY_PSA_MAILBOX_IRQN_ID) == (IRQn_Type)(-1)) {
```

Comment:
`-1` is oddly specific. What does the magic number map to?

Reply:
It refers to targets/TARGET_Cypress/TARGET_PSOC6_FUTURE/psoc6_utils.h:
```/** \brief Reserves particular NVIC channel if it is available.
 *
 * \param channel IRQn_Type     Channel to be reserved.
 * \param allocation_id uint32_t    Identifier.
 * \return  IRQ channel allocated or (-1) if no free channel is available.
 *
 */
IRQn_Type cy_m0_nvic_reserve_channel(IRQn_Type channel, uint32_t allocation_id);
```
Yeah, it's sloppy. But fixing this requires a global HAL implementation change, that I would rather prepare as a separate PR.